docs/topics/proc_starting-your-application-on-openshift-in-debugging-mode.adoc

```diff
@@ -0,0 +1,64 @@
+
+[#starting-your-application-on-openshift-in-debugging-mode_{context}]
+= Starting Your Application on OpenShift in Debugging Mode
+
+To debug your :runtime:-based application on OpenShift remotely, you must set the `JAVA_DEBUG` environment variable inside the container to `true` and configure port forwarding so that you can connect to your application from a remote debugger.
+
+.Prerequisites
+
+* Your application running on OpenShift.
+* The `oc` binary installed on your machine.
+
+.Procedure
+
+. Using the `oc` command, list the available deployment configurations:
++
+[source,bash]
+----
+$ oc get dc
+----
+
+. Set the `JAVA_DEBUG` environment variable in the deployment configuration of your application to `true`, which configures the JVM to open the port number 5005 for debugging. For example:
++
+[source,bash]
+----
+$ oc set env dc/my-application JAVA_DEBUG=true
+----
+
+. Redeploy the application if it is not set to redeploy automatically on environment change. For example:
++
+[source,bash]
+----
+$ oc rollout latest dc/my-application
+----
+
+. Configure port forwarding from the application pod:
```

Comment:
It's actually port forwarding from local machine _to_ application pod.

Reply:
This is a good point. The `JAVA_DEBUG` environment variable is specific to the Red Hat OpenJDK S2I image (and the Fabric8 Java images support it too), it's not a general-purpose thing.

On the other hand, general-purpose environment variables such as `_JAVA_OPTIONS` exist, see e.g. https://stackoverflow.com/questions/28327620/difference-between-java-options-java-tool-options-and-java-opts